frontend/src/stateless/index.ts

```diff
@@ -419,13 +419,83 @@ export async function fetchStatelessClusterKubeConfigs(dispatch: any) {
     });
 }
 
+/**
+ * deleteClusterKubeconfig deletes the kubeconfig for a stateless cluster from indexedDB
+ * @param clusterName
+ */
+export async function deleteClusterKubeconfig(clusterName: string): Promise<string | null> {
+  return new Promise<string | null>(async (resolve, reject) => {
+    try {
+      const request =
+        process.env.NODE_ENV === 'test'
+          ? indexedDBtest.open('kubeconfigs', 1)
+          : indexedDB.open('kubeconfigs', 1);
+
+      // The onupgradeneeded event is fired when the database is created for the first time.
+      request.onupgradeneeded = handleDatabaseUpgrade;
+
+      // The onsuccess event is fired when the database is opened.
+      // This event is where you specify the actions to take when the database is opened.
+      request.onsuccess = function handleDatabaseSuccess(event: DatabaseEvent) {
+        const db = event.target.result;
+        const transaction = db.transaction(['kubeconfigStore'], 'readwrite');
+        const store = transaction.objectStore('kubeconfigStore');
+
+        // The onsuccess event is fired when the request has succeeded.
+        // This is where you handle the results of the request.
+        // The result is the cursor. It is used to iterate through the object store.
+        // The cursor is null when there are no more objects to iterate through.
+        // The cursor is used to find the kubeconfig by cluster name.
+        store.openCursor().onsuccess = function storeSuccess(event: Event) {
+          // delete the kubeconfig by cluster name
+          const successEvent = event as CursorSuccessEvent;
+          const cursor = successEvent.target.result;
+          if (cursor) {
+            const kubeconfigObject = cursor.value;
+            const kubeconfig = kubeconfigObject.kubeconfig;
+
+            const parsedKubeconfig = jsyaml.load(atob(kubeconfig)) as KubeconfigObject;
```

Comment:
This could fail, so it probably needs a try/catch/reject.

Reply:
This whole is inside try catch here https://github.com/headlamp-k8s/headlamp/pull/1806/files#diff-c420684a8dc177e0c1311ea7c11cfd598ec94e9563b1ab6c095f14bdf0d602d8R428. Do you need more specific try catch is needed?